src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1856,8 +1857,11 @@ MasterAggregateExpression(Aggref *originalAggregate,
 		const char *aggregateName = AggregateNames[aggregateType];
 		Oid aggregateFunctionId = AggregateFunctionOid(aggregateName, workerReturnType);
 		Oid masterReturnType = get_func_rettype(aggregateFunctionId);
-
-		Aggref *newMasterAggregate = copyObject(originalAggregate);
+		if (masterReturnType == ANYELEMENTOID)
+		{
+			masterReturnType = workerReturnType;
```

Comment:
This line is not covered by a test, when is it used? Could also use a comment with an example.

Reply:
Fixed now